sqlx-core/src/query.rs

```diff
@@ -114,22 +114,26 @@ where
     /// The [`query_as`](super::query_as::query_as) method will construct a mapped query using
     /// a [`FromRow`](super::from_row::FromRow) implementation.
     #[inline]
-    pub fn map<F, O>(self, f: F) -> Map<'q, DB, impl TryMapRow<DB, Output = O>, A>
+    pub fn map<F, O>(
+        self,
+        mut f: F,
+    ) -> Map<'q, DB, impl Send + FnMut(DB::Row) -> Result<O, Error>, A>
     where
-        F: MapRow<DB, Output = O>,
+        F: Send + FnMut(DB::Row) -> O,
```

Comment:
Auto traits are usually listed last in the bounds unless this is what Clippy recommends now.

Reply:
I just find it easier to parse with `Fn*` being last. Will reverse it.

Reply:
Yeah it's not exactly great the other way but when skimming APIs I'm used to seeing the "most important" trait bound first.

Reply:
It looks like `thread::spawn` makes this cleaner by separating the bounds which is better for readability but makes the signature take up even more lines: https://doc.rust-lang.org/stable/std/thread/fn.spawn.html